channel2/reconnecting_dialer.go

```diff
@@ -42,6 +43,15 @@ func NewReconnectingDialer(identity *identity.TokenId, endpoint transport.Addres
 	}
 }
 
+func NewReconnectingDialerWithListener(identity *identity.TokenId, endpoint transport.Address, headers map[int32][]byte, reconnectListener func()) UnderlayFactory {
+	return &reconnectingDialer{
+		identity:          identity,
+		endpoint:          endpoint,
+		headers:           headers,
+		reconnectListener: reconnectListener,
```

Comment:
Kind of feel like this should be called `reconnectHandler` instead of `reconnectListener`. The word `listener` is already overloaded in `channel2`, and what's here is more of a "handler" than a listener, in my opinion.

Reply:
:+1: fixed